apko: 0.27.9 -> 0.29.1

06kellyjac opened the pull request:
Diff: https://github.com/chainguard-dev/apko/compare/refs/tags/v0.27.9...refs/tags/v0.29.1

Changelog: https://github.com/chainguard-dev/apko/blob/main/NEWS.md

* Use finalAttrs
* Reduce use of with lib
* Move to by-name and remove from all-packages
* Skip tests on darwin and use writableTmpDirAsHomeHook
* Fix completion generation for cross-compilation


<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [Nixpkgs 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/doc/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/doc/manual/release-notes/rl-2505.section.md) Nixpkgs Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
- [NixOS 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) NixOS Release notes)
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md), [pkgs/README.md](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md), [maintainers/README.md](https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md) and other contributing documentation in corresponding paths.

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

06kellyjac commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 421282`

Logs: https://github.com/06kellyjac/nixpkgs-review-gha/actions/runs/15977879406


---
### `x86_64-linux` (sandbox = true)
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>apko</li>
  </ul>
</details>


---
### `aarch64-linux` (sandbox = true)
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>apko</li>
  </ul>
</details>


---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>apko</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>apko</summary>
<pre>testing.tRunner.func1.2({0x446db40, 0xc0003eb060})
	/nix/store/n0gqcdpdk0ap2ikasi36r0szwpk2g35q-go-1.24.4/share/go/src/testing/testing.go:1734 +0x21c
testing.tRunner.func1()
	/nix/store/n0gqcdpdk0ap2ikasi36r0szwpk2g35q-go-1.24.4/share/go/src/testing/testing.go:1737 +0x35e
panic({0x446db40?, 0xc0003eb060?})
	/nix/store/n0gqcdpdk0ap2ikasi36r0szwpk2g35q-go-1.24.4/share/go/src/runtime/panic.go:792 +0x132
net/http/httptest.newLocalListener()
	/nix/store/n0gqcdpdk0ap2ikasi36r0szwpk2g35q-go-1.24.4/share/go/src/net/http/httptest/server.go:71 +0xda
net/http/httptest.NewUnstartedServer(...)
	/nix/store/n0gqcdpdk0ap2ikasi36r0szwpk2g35q-go-1.24.4/share/go/src/net/http/httptest/server.go:119
net/http/httptest.NewServer({0x46aef20, 0xc0000b1240})
	/nix/store/n0gqcdpdk0ap2ikasi36r0szwpk2g35q-go-1.24.4/share/go/src/net/http/httptest/server.go:106 +0x25
chainguard.dev/apko/internal/cli_test.TestPublish(0xc000748700)
	/private/tmp/nix-build-apko-0.28.0.drv-0/source/internal/cli/publish_test.go:56 +0xdf
testing.tRunner(0xc000748700, 0x46a4948)
	/nix/store/n0gqcdpdk0ap2ikasi36r0szwpk2g35q-go-1.24.4/share/go/src/testing/testing.go:1792 +0xf4
created by testing.(*T).Run in goroutine 1
	/nix/store/n0gqcdpdk0ap2ikasi36r0szwpk2g35q-go-1.24.4/share/go/src/testing/testing.go:1851 +0x413
FAIL	chainguard.dev/apko/internal/cli	0.142s
FAIL</pre>
</details>
</details>


---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>apko</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>apko</summary>
<pre>testing.tRunner.func1.2({0x104f43a20, 0x1400049cb30})
	/nix/store/rq7irijkj3nhapmjcv9d96xgkisj55x2-go-1.24.4/share/go/src/testing/testing.go:1734 +0x1ac
testing.tRunner.func1()
	/nix/store/rq7irijkj3nhapmjcv9d96xgkisj55x2-go-1.24.4/share/go/src/testing/testing.go:1737 +0x334
panic({0x104f43a20?, 0x1400049cb30?})
	/nix/store/rq7irijkj3nhapmjcv9d96xgkisj55x2-go-1.24.4/share/go/src/runtime/panic.go:792 +0x124
net/http/httptest.newLocalListener()
	/nix/store/rq7irijkj3nhapmjcv9d96xgkisj55x2-go-1.24.4/share/go/src/net/http/httptest/server.go:71 +0xdc
net/http/httptest.NewUnstartedServer(...)
	/nix/store/rq7irijkj3nhapmjcv9d96xgkisj55x2-go-1.24.4/share/go/src/net/http/httptest/server.go:119
net/http/httptest.NewServer({0x105184ba0, 0x140003869c0})
	/nix/store/rq7irijkj3nhapmjcv9d96xgkisj55x2-go-1.24.4/share/go/src/net/http/httptest/server.go:106 +0x28
chainguard.dev/apko/internal/cli_test.TestPublish(0x140002cb180)
	/private/tmp/nix-build-apko-0.28.0.drv-0/source/internal/cli/publish_test.go:56 +0xb8
testing.tRunner(0x140002cb180, 0x10517a5c8)
	/nix/store/rq7irijkj3nhapmjcv9d96xgkisj55x2-go-1.24.4/share/go/src/testing/testing.go:1792 +0xe4
created by testing.(*T).Run in goroutine 1
	/nix/store/rq7irijkj3nhapmjcv9d96xgkisj55x2-go-1.24.4/share/go/src/testing/testing.go:1851 +0x374
FAIL	chainguard.dev/apko/internal/cli	0.095s
FAIL</pre>
</details>
</details>

emilylange reviewed (changes requested):
- 0.27.9 -> 0.29.0
- Use finalAttrs
- Reduce use of with lib
- Move to by-name and remove from all-packages
- Skip tests on darwin

This should be separate commits for each logical unit, as per our contributing guidelines. Please also add the reason for disabling the unit tests on darwin to the commit message body.


https://github.com/NixOS/nixpkgs/blob/88f45f110b80aa3ea6b2e3fb3c168984e47e296a/CONTRIBUTING.md#commit-conventions

06kellyjac committed:
apko: 0.27.9 -> 0.29.1

Diff: chainguard-dev/apko@refs/tags/v0.27.9...refs/tags/v0.29.1

Changelog: https://github.com/chainguard-dev/apko/blob/main/NEWS.md

06kellyjac committed:
apko: move to pkgs/by-name

06kellyjac committed:
apko: remove pin on buildGoModules version

06kellyjac committed:
apko: move to finalAttrs pattern

06kellyjac committed:
apko: fix tests on darwin and use writableTmpDirAsHomeHook

Skip tests on darwin due to some local networking failures
`__darwinAllowLocalNetworking = true;` wasn't sufficient for
aarch64 or x86_64

06kellyjac committed:
apko: move version check to versionCheckHook

06kellyjac committed:
apko: reduce usage of `with lib;`

06kellyjac committed:
apko: adjust completion generation to work when cross-compiling

emilylange reviewed (approved):
Thanks for splitting at least most of the changes into separate commits.

Just one more thing I want to say for completeness’s sake. I won't block on that and will merge this later today either way:

Your commit message body (not title) generated by `nixpkgs-update` in 52c39cbac8fa101fcc91bb8cc5c8bd3dd8fa1dda is effectively worthless now. When you split and redid your commits you appear to have messed up copying the previous auto-generated diff link (it is no longer a link) and the changelog url isn't of any value either, as that `NEWS.md` hasn't been updated in 1½ years and upstream has since moved to exclusively listing the changes in their GitHub releases.
